### PR TITLE
DAOS-7821 gha: Perform fewer builds for pull requests.

### PR DESCRIPTION
**Pull request by ashleypittman**
Cut down the number of distros that we build for PRs, remove ci.yml
completely.

Now we're building on master again we don't need to cover so many
distros with PRs, and github has been queueing a number of our
builds recently.

Set the target branch in the hash key for docker images.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: .github/workflows/ci2.yml [hunk 1]
```diff
@@ -62,33 +62,21 @@ jobs:
...
-            with: rockylinux/rockylinux:8
-            java: YES
-          - distro: alma
-            base: centos.8
-            with: almalinux:8
-            java: YES
+            java: no
           - distro: centos.8
             base: centos.8
             with: centos:8
```
Comment (johannlombardi) [inline .github/workflows/ci2.yml:74]: Let's drop centos.8 here and use Rocky Linux instead.

Comment (ashleypittman) [reply .github/workflows/ci2.yml]: Done, testing it now.  I ordered the matrix entries at the same time.

File: ci/gha_helper.py [hunk 2]
```diff
@@ -37,8 +50,14 @@ def main():
...
     cmd.extend(BUILD_FILES)
 
+    # Check that there are no typos in the BUILD_FILES, and that it's kept up-to-date.
+    for fname in BUILD_FILES:
+        assert os.path.exists(fname)
+
     if base_distro:
-        cmd.append('utils/docker/Dockerfile.{}'.format(base_distro))
+        dockerfile = 'utils/docker/Dockerfile.{}'.format(base_distro)
+        assert os.path.exists(dockerfile)
```
Comment (ashleypittman) [inline ci/gha_helper.py:59]: This needs updating to non-centos8 distros using the centos dockerfile

**Review by mjmac — APPROVED** [review]
It seems reasonable to me. The point of the key updates is to try to improve reuse of common docker layers, right?

**Comment by ashleypittman** [comment]
> It seems reasonable to me. The point of the key updates is to try to improve reuse of common docker layers, right?

Lots of changes here, this is refining an equally big update from about 10 days ago but:

1. Perform fewer builds for PRs to prevent long queues.
2. Make hash reuse (for docker layers) more likely by putting both the branch in the hash name, but also checking the last two revisions.
3. Put fewer distros in the hash to reduce churn/eviction.
4. Do not refresh images from upstream in PRs, so reduce load and failure rate.

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**